storage/oasis/nodeapi/file/consensus.go

```diff
@@ -0,0 +1,226 @@
+package file
+
+import (
+	"context"
+
+	"github.com/akrylysov/pogreb"
+	"github.com/fxamacker/cbor"
+	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	coreCommon "github.com/oasisprotocol/oasis-core/go/common"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	genesis "github.com/oasisprotocol/oasis-core/go/genesis/api"
+
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi"
+)
+
+// FileConsensusApiLite provides access to the consensus API of an Oasis node.
+// Since FileConsensusApiLite is backed by a file containing the cached responses
+// to `ConsensusApiLite` calls, this data is inherently compatible with the
+// current indexer and can thus handle heights from both Cobalt/Damask.
+type FileConsensusApiLite struct {
+	db           pogreb.DB
+	consensusApi nodeapi.ConsensusApiLite
+}
+
+var _ nodeapi.ConsensusApiLite = (*FileConsensusApiLite)(nil)
+
+func NewFileConsensusApiLite(filename string, consensusApi nodeapi.ConsensusApiLite) (*FileConsensusApiLite, error) {
+	db, err := pogreb.Open(filename, &pogreb.Options{BackgroundSyncInterval: -1})
+	if err != nil {
+		return nil, err
+	}
+	return &FileConsensusApiLite{
+		db:           *db,
+		consensusApi: consensusApi,
+	}, nil
+}
+
+func (c *FileConsensusApiLite) get(key []byte, result interface{}) error {
+	res, err := c.db.Get(key)
+	if err != nil {
+		return err
+	}
+	return cbor.Unmarshal(res, result)
+}
```

Comment:
hrm `cbor` complains with `cbor: invalid map key type: slice` due to unmarshalling into the `result` interface here.

Reply:
this was that event `Body interface{}` problem right?

Reply:
yep